javaparser-core/src/main/java/com/github/javaparser/ast/validator/language_level_validations/chunks/FlexibleConstructorValidator.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (C) 2007-2010 Júlio Vilmar Gesser.
+ * Copyright (C) 2011, 2013-2025 The JavaParser Team.
+ *
+ * This file is part of JavaParser.
+ *
+ * JavaParser can be used either under the terms of
+ * a) the GNU Lesser General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ * b) the terms of the Apache License
+ *
+ * You should have received a copy of both licenses in LICENCE.LGPL and
+ * LICENCE.APACHE. Please refer to those files for details.
+ *
+ * JavaParser is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ */
+
+package com.github.javaparser.ast.validator.language_level_validations.chunks;
```

Comment:
Could you remove the ‘chunk’ package and integrate the content of this class into the validator (as is done for the other validators)?

Reply:
I agree with this idea. If the code for a validator method is too complex to reasonably include in the validator class itself, then I think the validation logic should be reconsidered. It may make sense in some cases, but most validator should just check whether a flag is set or a simple condition (in this case whether any statements appear before the constructor invocation)